### PR TITLE
[issue - 493]Organization lock incorrectly validates against valid repos if org name is mixed case #493

### LINKED ISSUE #493: GitHub 2.1.0 - Organization lock incorrectly validates against valid repos if org name is mixed case
It seems the organization lock functionality has a validation issue. 
When organization lock is used to lock subscriptions to a mixed case org such as `DHaussermann` it is no longer possible to add subscriptions even when they are valid. 
This issue does not occur for an organization such as `mattermost` where it seems to add subscriptions 

Note that `https://github.com/DHaussermann` is not actually an Organization in the GitHub sense but I don't believe this is related to the issue as the locking subscriptions to DHaussermann worked in the previous release. 
image
image

**Note** There is a work-around for this which is to save the org name in all lower case but, this is not discoverable and existing installations would be impacted.

Steps:
- Build GitHub 2.1.0 from the version bump PR https://github.com/mattermost/mattermost-plugin-github/pull/490 
- Install and configure GitHub Plugin
- Set _Organization lock_ to `DHaussermann`  
- Attempt to subscribe to this public repo https://github.com/DHaussermann/hello-world - type `/github subscriptions add DHaussermann/hello-world`
**Observed**: `organization not supported: only repositories in the DHaussermann organization are supported`

### PR DESCRIPTION
**Pull request by sibasankarnayak**
Removed the convert logic which converts repo and org to lowercase

ticket here

Fixes #493

### CODE REVIEW

File: server/plugin/subscriptions.go [hunk 1]
```diff
@@ -114,9 +114,6 @@ func (p *Plugin) Subscribe(ctx context.Context, githubClient *github.Client, use
 		return errors.Errorf("invalid repository")
 	}
 
-	owner = strings.ToLower(owner)
-	repo = strings.ToLower(repo)
```
[the remark this answers is not in the document]
Comment (sibasankarnayak) [reply server/plugin/subscriptions.go:118]: @hanzei can you look into this

[the remark this answers is not in the document]
Comment (hanzei) [reply server/plugin/subscriptions.go:118]: I assume the GitHub api is case insensitive. Can you confirm that this is the case?

If so, yes, my suggestion is to case them to lower case when we store the information.

[the remark this answers is not in the document]
Comment (sibasankarnayak) [reply server/plugin/subscriptions.go:118]: @hanzei checked this issue occurs as github api are case sensitive, have checked it and confirmed the same

[the remark this answers is not in the document]
Comment (hanzei) [reply server/plugin/subscriptions.go:118]: > @hanzei checked this issue occurs as github api are case sensitive, have checked it and confirmed the same

What API did you test that with? Is there a source that you can cite? 

I've tested with some of the label endpoint and there the repo and org are case insensitive. The URL's are also case insensitive: e.g. with https://github.com/Mattermost

**Review by hanzei — APPROVED** [review]
LGTM

**Comment by dipak-demansol** [comment]
@dylan @hanzei if the user use full URL to subscribe repo as ex "https://github.com/DHaussermann/hello-world" and Some How if the URL Contain specifically capital test as this "https://GITHUB.COM" or "HTTPS://" then subscription will not work so is it right? can we ignore this?

File: server/plugin/plugin.go [hunk 2]
```diff
@@ -680,7 +680,7 @@ func (p *Plugin) checkOrg(org string) error {
 	config := p.getConfiguration()
 
 	configOrg := strings.TrimSpace(config.GitHubOrg)
-	if configOrg != "" && configOrg != org {
+	if configOrg != "" && configOrg != org && strings.ToLower(configOrg) != org {
```
Comment (mickmister) [inline server/plugin/plugin.go:683]: Should we also cast `org` to lower?

**Review by mickmister — APPROVED** [review]

File: server/plugin/plugin.go (hunk 2 above)
Comment (sibasankarnayak) [reply server/plugin/plugin.go:683]: @mickmister We have already did it in previous function

**Review by dipak-demansol — APPROVED** [review]
**LGTM**

**Comment by hanzei** [comment]
> if the user use full URL to subscribe repo as ex "[DHaussermann/hello-world](https://github.com/DHaussermann/hello-world)" and Some How if the URL Contain specifically capital test as this "https://GITHUB.COM" or "HTTPS://" then subscription will not work **so is it right?** **or can we ignore this?**

Capitalized URL's are out of the scope of this PR.

**Pull request merged**